src/puppetlabs/puppetdb/cli/services.clj

```diff
@@ -208,14 +208,20 @@
   being fully started when PuppetDB starts. This connection pool will
   be opened and closed within the body of this function."
   [write-db-config config]
-  (with-open [init-db-pool (jdbc/make-connection-pool
-                            (assoc write-db-config
-                                   ;; Block waiting to grab a connection
-                                   :connection-timeout 0
-                                   ;; Only allocate connections when needed
-                                   :pool-availability-threshold 0))]
-    (let [db-pool-map {:datasource init-db-pool}]
-      (initialize-schema db-pool-map config))))
+  (loop [db-spec (assoc write-db-config
+                        ;; Block waiting to grab a connection
+                        :connection-timeout 15000
+                        :pool-name "PDBMigrationsPool")]
+    (if-let [result
+             (try
+               (with-open [init-db-pool (jdbc/make-connection-pool db-spec)]
+                 (let [db-pool-map {:datasource init-db-pool}]
+                   (initialize-schema db-pool-map config)
+                   ::success))
+               (catch java.sql.SQLTransientConnectionException e
+                 nil))]
```

Comment:
So we only really see this logged at INFO level when it goes wrong:

```
2015-12-09 13:40:12,434 INFO  [main] [c.z.h.HikariDataSource] PDBMigrationsPool - is starting.
2015-12-09 13:38:22,302 INFO  [main] [c.z.h.p.HikariPool] PDBMigrationsPool - is closing down.
```

Unless the user is running DEBUG, then they won't see the persistent retries (which is a good thing) but they also won't see the failure reason ever, probably doesn't leave enough clues for the user. Perhaps we need to catch the exception and log it here at ERROR or something like that?